src/backend/aspen/api/utils/fasta_streamer.py

```diff
@@ -0,0 +1,103 @@
+import re
+from enum import Enum
+from typing import AsyncGenerator, Optional, Set
+
+import sqlalchemy as sa
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import joinedload
+
+from aspen.api.utils import authz_samples_cansee
+from aspen.database.models import DataType, Sample, UploadedPathogenGenome
+from aspen.database.models.usergroup import User
+
+
+class SpecialtyDownstreams(Enum):
+    """Canonical internal/external names for downstreams that require special logic."""
+
+    USHER = "USHER"
+
+
+class FastaStreamer:
+    def __init__(
+        self,
+        db: AsyncSession,
+        user: User,
+        sample_ids: Set[str],
+        downstream_consumer: Optional[str] = None,
+    ):
+        self.db = db
+        self.user = user
+        self.cansee_groups_private_identifiers: Set[int] = {
+            cansee.owner_group_id
+            for cansee in user.group.can_see
+            if cansee.data_type == DataType.PRIVATE_IDENTIFIERS
+        }
+        # query for samples
+        all_samples_query = sa.select(Sample).options(  # type: ignore
+            joinedload(Sample.uploaded_pathogen_genome, innerjoin=True).undefer(  # type: ignore
+                UploadedPathogenGenome.sequence
+            ),
+        )
+        # Enforce AuthZ
+        self.authz_samples_query = authz_samples_cansee(
```

Comment:
The original FastaStreamer was using `authz_sample_filters` which has a slightly different cansee filter. Are we changing this intentionally?